cpp/src/arrow/dataset/file_csv_test.cc

```diff
@@ -36,7 +37,21 @@
 namespace arrow {
 namespace dataset {
 
-class TestCsvFileFormat : public testing::TestWithParam<Compression::type> {
+class ArrowCsvWriterMixin {
```

Comment:
Naming nit: Not sure `mixin` is the right term here.  When I think `mixin` I think multiple inheritance.  Maybe `Strategy`?  Also, why "ArrowCsv".  Are there other CSV writers?  Also, maybe there is a way to use composition here?  Have an abstract method in `FileFormatFixtureMixin` that returns a `FormatStrategy` or something?  I know "prefer composition over inheritance" but I don't know if there is a similar "prefer composition over template" guideline :laughing:

Reply:
I originally tried with abstract methods, but then because of the inheritance tree, you'd have to re-implement them in the AbcFileFormatTest and in the AbcFileFormatScanTest fixtures (I had issues when trying to just have the latter inherit the former). Probably there is some combination of superclasses and interfaces that'll make it work but at that point I decided a template was the lesser evil.